tests/acceptance/comparative_booster_study/dataset_definition.py

```diff
@@ -146,31 +147,23 @@ def has_prior_meds(codelist, where=True):
 
 dataset.sex = schema.patients.sex
 
-#    # https://github.com/opensafely/risk-factors-research/issues/51
-#    bmi=patients.categorised_as(
-#      {
-#        "Not obese": "DEFAULT",
-#        "Obese I (30-34.9)": """ bmi_value >= 30 AND bmi_value < 35""",
-#        "Obese II (35-39.9)": """ bmi_value >= 35 AND bmi_value < 40""",
-#        "Obese III (40+)": """ bmi_value >= 40 AND bmi_value < 100""",
-#        # set maximum to avoid any impossibly extreme values being classified as obese
-#      },
-#      bmi_value=patients.most_recent_bmi(
-#        on_or_after="covid_vax_disease_3_date - 5 years",
-#        minimum_age_at_measurement=16
-#      ),
-#      return_expectations={
-#        "rate": "universal",
-#        "category": {
-#          "ratios": {
-#            "Not obese": 0.7,
-#            "Obese I (30-34.9)": 0.1,
-#            "Obese II (35-39.9)": 0.1,
-#            "Obese III (40+)": 0.1,
-#          }
-#        },
-#      },
-#    ),
+# BMI
+# https://github.com/opensafely/risk-factors-research/issues/51
+bmi_measurement = most_recent_bmi(
+    # This isn't _exactly_ 5 years as the old study used, but I can't see that would
+    # matter here
+    where=events.date.is_after(baseline_date.subtract_days(5 * 365)),
```

Comment:
Is there an argument for a `.subtract_years` method?

Reply:
Yes, I think there is. I've actually got the sketch of ticket for this, but there's a little bit of complexity here because we need to handle leap years consistently and different databases have different behaviour here (this is one of the reasons spec tests are great).

And then I realised all that precision is totally spurious in this particular case because the dates of birth are all rounded anyway, so here were may as well do the simple but inexact thing.

But I still think it would be worth rounding out our suite of available date functions. I'll turn it into a proper ticket at some stage.